sdk/tests/program_macros.rs

```diff
@@ -0,0 +1,207 @@
+use bincode::serialize;
+use serde_derive::{Deserialize, Serialize};
+use solana_sdk::{
+    instruction::{AccountMeta, Instruction},
+    pubkey::Pubkey,
+};
+use solana_sdk_program_macros::instructions;
+
+#[instructions(test_program::id())]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum TestInstruction {
+    /// Transfer lamports
+    #[accounts(
+        from_account(SIGNER, WRITABLE, desc = "Funding account"),
```

Comment:
Looking back at the proposal, I see this syntax made it in, but I still find it really confusing. Remind me why we can't write `(from_account, SIGNER | WRITABLE, desc = "Funding account")`?

Reply:
This is using the standard attribute syntax for a list (of accounts/names), which each carry a list of account attributes.
In addition to the "standardization" aspect, it's significantly easier to parse the attributes in the macro if they can be mapped into `syn` library meta types.

But if you feel really strongly about this, I'll write a custom parser.

One thing that I'm not crazy about with your proposed syntax is the strict ordering that's required inside the parens in order to identify the account name. As it is, the account name is unambiguous, because it is the item `Path`, and account attributes inside the parens can be listed in any order and remain parsable.

Reply:
I'm looking for as little custom parsing as possible and that wherever possible, pass in Rust expressions. I don't want to learn a new DSL to annotate this enum.

Reply:
Maybe this is possible?

```
use solana_sdk_program_macros::{SIGNER, WRITABLE};
...
#[accounts(
       (from_account, SIGNER | WRITABLE, desc: "Funding account")
  )]
```

So a tuple of an identifier, an expression that evaluates to a u64, and then key-value pairs that use the struct field syntax.

Reply:
Sure, it's possible. I think this is approx the same amount of extra handling as your previous version.

I'm not sure whether it's easy to actually evaluate the `SIGNER | WRITABLE` expression in the macro. I'll have to look into that.